### PR TITLE
elasticsearch: logging with Lumberjack.

### PR DESCRIPTION
**Pull request by jmacmahon**
* Added disabled-by-default config options for Elasticsearch logging.
* Replaced rnkPAGEVIEWS and rnkDOWNLOADS with Elasticsearch.

Signed-off-by: Joe MacMahon <joe.macmahon@cern.ch>

### CODE REVIEW

**Comment by tiborsimko** [comment]
@kasioumis @egabancho Please add wanted milestone.

**Comment by egabancho** [comment]
@jmacmahon could you please make your PR on top of Invenio master and not CDS production branch?

**Comment by jmacmahon** [comment]
Yes, sorry about that, I forgot this PR even existed and rebased my elasticsearch_logging branch on top of CDS/prod to make things easier for production...

**Comment by kaplun** [comment]
> @jmacmahon could you please make your PR on top of Invenio master and not CDS production branch?

A bit OT, but... are all the referred commits from @pglauner on the search engine/solr already integrated into Invenio master?

**Comment by tiborsimko** [comment]
> A bit OT, but... are all the referred commits from @pglauner on the search engine/solr already integrated into Invenio master?

Yes, according to my understanding, except for PR #2235 that touches test suite and needs confirming with Solr option switched on.

**Comment by egabancho** [comment]
@kaplun you can check what we have on our [ops branch]( https://github.com/CERNDocumentServer/cds-ops/commits/prod), there are a few things from Patrick that need some ... refactoring

File: modules/bibdocfile/lib/bibdocfile.py [hunk 1]
```diff
@@ -2856,12 +2863,31 @@ def register_download(self, ip_address, version, docformat, userid=0, recid=0):
...
+            # TODO: Move to CFG_ variables
+            BOTS = ['Googlebot', 'bingbot']
+            BOT_TTL = '30d'
+            for bot in BOTS:
+                if user_agent.find(bot) != -1:
+                    log_entry['_ttl'] = BOT_TTL
+                    break
+            _DOWNLOAD_LOG.info(log_entry)
+        else:
+            return run_sql("INSERT DELAYED INTO rnkDOWNLOADS "
```
Comment (egabancho) [inline modules/bibdocfile/lib/bibdocfile.py]: Remember to remove the `DELAYED`, see 386ac9d5ef1041f1ac98725e0c3aea178d811f96

File: requirements.txt [hunk 2]
```diff
@@ -29,3 +29,4 @@ redis==2.9.0
 nydus==0.10.6
 Cerberus==0.5
 matplotlib
+lumberjack==git+https://github.com/CERNDocumentServer/lumberjack.git
```
Comment (egabancho) [inline requirements.txt]: can you update it as in https://github.com/CERNDocumentServer/invenio/pull/89?

**Comment by jirikuncar** [comment]
Invenio 1.x is in feature freeze state.

**Pull request closed without merging**